aztk/spark/client.py

```diff
@@ -131,3 +136,124 @@ def get_application_status(self, cluster_id: str, app_name: str):
+    def submit_job(self, job_configuration):
+        try:
+            zip_resource_files = upload_node_scripts.zip_scripts(self.blob_client,
+                                                                 job_configuration.id,
+                                                                 job_configuration.custom_scripts,
+                                                                 job_configuration.spark_configuration)
+            
+            start_task = create_cluster_helper.generate_cluster_start_task(self,
+                                                                           zip_resource_files,
+                                                                           job_configuration.gpu_enabled,
+                                                                           job_configuration.docker_repo)
+             
+            application_tasks = []
+            for application in job_configuration.applications:
+                application_tasks.append((application, submit_helper.generate_task(self, job_configuration.id, application)))
+
+            job_manager_task = job_submit_helper.generate_task(self, job_configuration, application_tasks)
+
+
+            software_metadata_key = "spark"
+
+            vm_image = models.VmImage(
+                publisher='Canonical',
+                offer='UbuntuServer',
+                sku='16.04')
+            
+            if job_configuration.max_dedicated_nodes and not job_configuration.max_low_pri_nodes:
+                autoscale_formula = "maxNumberofVMs = {0}; targetNumberofVMs = {1}; $TargetDedicatedNodes=min(maxNumberofVMs, targetNumberofVMs)".format(
+                    job_configuration.max_dedicated_nodes, job_configuration.max_dedicated_nodes)
+            elif job_configuration.max_low_pri_nodes and not job_configuration.max_dedicated_nodes:
+                autoscale_formula = "maxNumberofVMs = {0}; targetNumberofVMs = {1}; $TargetLowPriorityNodes=min(maxNumberofVMs, targetNumberofVMs)".format(
+                    job_configuration.max_low_pri_nodes, job_configuration.max_low_pri_nodes)
+            else:
+                raise error.AztkError("Jobs do not support both dedicated and low priority nodes. JobConfiguration fields max_dedicated_nodes and max_low_pri_nodes are mutually exclusive values.")
+            
+            
+            job = self.__submit_job(
+                job_configuration=job_configuration,
+                start_task=start_task,
+                job_manager_task=job_manager_task,
+                autoscale_formula=autoscale_formula,
+                software_metadata_key=software_metadata_key,
+                vm_image_model=vm_image)
+        
+        except batch_error.BatchErrorException as e:
+            raise error.AztkError(helpers.format_batch_exception(e))
+
+    def list_jobs(self):
+        try:
+            return [models.Job(cloud_job_schedule) for cloud_job_schedule in self.batch_client.job_schedule.list()]
```

Comment:
Is this correct? You're asking to list jobs, but returning job schedules... Is the expectation that you can get a job from a schedule elsewhere?

Reply:
Also, curious as far as coding pattern goes, that all the other methods proxy through job_submit_helper.foo() while this one hits the batch API directly. Any reason in particular?